lib/song.rb

```diff
@@ -0,0 +1,58 @@
+class Song
+  extend Concerns::Findable
+
+  attr_accessor :name
+  attr_reader :artist
+  attr_reader :genre
+
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
+    self.artist = artist
+    self.genre = genre
+  end
+
+  def artist=(artist)
+    @artist = artist
+    artist && artist.add_song(self)
+  end
+
+  def genre=(genre)
+    @genre = genre
+    genre && (genre.songs << self unless genre.songs.include?(self))
```

Comment:
#### Code Quality - Critical
Too much is happening on this line, making it hard to read. This can be helped by not having to check for nil here, and instead checking for nil during initialization.